packages/common/src/TransactionUtils.js

```diff
@@ -0,0 +1,128 @@
+
+  // Next, simulate transaction and also extract return value if its a state-modifying transaction. If the function is state
+  // modifying, then successfully sending it will return the transaction receipt, not the return value, so we grab it here.
+  let returnValue, estimatedGas;
+  try {
+    [returnValue, estimatedGas] = await Promise.all([
+      transaction.call({ from: transactionConfig.from }),
+      transaction.estimateGas({ from: transactionConfig.from })
+    ]);
+  } catch (error) {
+    error.type = "call";
+    throw error;
+  }
+
+  // .call() succeeded, now broadcast transaction.
+  try {
+    transactionConfig = { ...transactionConfig, gas: Math.floor(estimatedGas * GAS_LIMIT_BUFFER) };
+
+    // ynatm doubles gasPrice every retry. Tries every minute (and increases gas price according to DOUBLE method) if tx
+    // hasn't mined. Min Gas price starts at caller's transactionConfig.gasPrice, with a max gasPrice of x6.
+    const gasPriceScalingFunction = ynatm.DOUBLES;
+    const retryDelay = 60000;
+    const minGasPrice = transactionConfig.gasPrice;
+    const maxGasPrice = 2 * 3 * minGasPrice;
+
+    const receipt = await ynatm.send({
+      sendTransactionFunction: gasPrice => transaction.send({ ...transactionConfig, gasPrice }),
+      minGasPrice,
+      maxGasPrice,
+      gasPriceScalingFunction,
+      delay: retryDelay
+    });
+
+    return { receipt, returnValue, transactionConfig };
+  } catch (error) {
+    error.type = "send";
+    throw error;
+  }
+};
+
+/**
+ * Checks if an account has a pending transaction.
+ * @param {*Object} web3.js object for making queries and accessing Ethereum related methods.
+ * @param {*string} account account to check.
+ * @return Bool true if the account has pending transaction and false if no pending transaction.
+ */
+const accountHasPendingTransactions = async (web3, account) => {
+  const [currentMindedTransactions, currentTransactionsIncludingPending] = await Promise.all([
+    web3.eth.getTransactionCount(account, "latest"),
+    getPendingTransactionCount(web3, account)
+  ]);
+  return currentTransactionsIncludingPending > currentMindedTransactions;
+};
+
+/**
+ * Returns the number of pending transaction an account has. This method uses `provider.send` syntax. This undocumented
+ * web3 method lets you call direct jsonrpc methods on the provider. This method differs from the web3.js getTransactionCount
+ * by including mempool transactions. see https://infura.io/docs/ethereum/json-rpc/eth-getTransactionCount
+ * @param {*Object} web3.js object for making queries and accessing Ethereum related methods.
+ * @param {*string} account account to check.
+ * @returns number representing the number of transactions, including pending.
+ */
+const getPendingTransactionCount = async (web3, account) => {
+  const sendRpc = util.promisify(web3.currentProvider.send).bind(web3.currentProvider);
```

Comment:
So this will only work with infura providers? Wb alchemy for example, or other default nodes used in Polygon

Reply:
not actually 100% sure. I will test it out on cloud flare (our backup node) to double check. it is a normal web3 endpoint that should work with any web3 provider, in theory.